# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/config/gfxConfig.cpp
# Commit: d158b20d9a65
# Full Hash: d158b20d9a65cd05dd69a586473216b800523083
# Author: Dan Glastonbury <dan.glastonbury@gmail.com>
# Date: 2020-10-21 09:51:28
# Regressor Bug: 1595994
# File Overlap Count: 1
# Description:
#   Bug 1595994 - PD: Allow WMF decoding inside RDD process. r=mattwoodrow
#   
#   This assumes that access to Windows WMF and DirectX is allowed from RDD process.
#   
#   Depends on D91699
# ==============================================================================

diff -r d0c9c64867cc -r d158b20d9a65 gfx/config/gfxConfig.cpp
--- a/gfx/config/gfxConfig.cpp	Tue Oct 20 23:26:49 2020 +0000
+++ b/gfx/config/gfxConfig.cpp	Tue Oct 20 23:26:52 2020 +0000
@@ -153,6 +153,37 @@
 }
 
 /* static */
+void gfxConfig::Inherit(EnumSet<Feature> aFeatures,
+                        const DevicePrefs& aDevicePrefs) {
+  for (Feature feature : aFeatures) {
+    FeatureStatus status = FeatureStatus::Unused;
+    switch (feature) {
+      case Feature::HW_COMPOSITING:
+        status = aDevicePrefs.hwCompositing();
+        break;
+      case Feature::D3D11_COMPOSITING:
+        status = aDevicePrefs.d3d11Compositing();
+        break;
+      case Feature::OPENGL_COMPOSITING:
+        status = aDevicePrefs.oglCompositing();
+        break;
+      case Feature::ADVANCED_LAYERS:
+        status = aDevicePrefs.advancedLayers();
+        break;
+      case Feature::DIRECT2D:
+        status = aDevicePrefs.useD2D1();
+        break;
+      case Feature::WEBGPU:
+        status = aDevicePrefs.webGPU();
+        break;
+      default:
+        break;
+    }
+    gfxConfig::Inherit(feature, status);
+  }
+}
+
+/* static */
 bool gfxConfig::UseFallback(Fallback aFallback) {
   return sConfig->UseFallbackImpl(aFallback);
 }